src/iidm/VoltageLevel.cpp

```diff
@@ -262,6 +271,12 @@ VoltageLevel& VoltageLevel::setLowVoltageLimit(double lowVoltageLimit) {
     return *this;
 }
 
+void VoltageLevel::setNetworkRef(Network& network) {
+    if (static_cast<bool>(m_networkRef)) {
```

Comment:
Why this test? m_networkRef is always set except if the VoltageLevel is detached. Then, you will never enter this method. Do I miss something?